### PR TITLE
Derive `PartialEq` and `Clone` for some structs

### PR DESCRIPTION
**Pull request by TianyiShi2001**

### CODE REVIEW

File: src/structs/atom.rs [hunk 1]
```diff
@@ -473,3 +473,4 @@ impl PartialEq for Atom {
             && self.b_factor == other.b_factor
     }
 }
+impl Eq for Atom {}
```
Comment (nonnominandus) [inline src/structs/atom.rs:476]: As Atom contains multiple f64 values it cannot implement Eq as f64 does not implement Eq. [See this question](https://users.rust-lang.org/t/what-is-the-difference-between-eq-and-partialeq/15751)

File: src/structs/chain.rs [hunk 2]
```diff
@@ -345,3 +345,5 @@ impl PartialEq for Chain {
         self.id() == other.id() && self.residues == other.residues
     }
 }
+
+impl Eq for Chain {}
```
Comment (nonnominandus) [inline src/structs/chain.rs:349]: As Chain contains multiple f64 values it cannot implement Eq as f64 does not implement Eq.

File: src/structs/model.rs [hunk 3]
```diff
@@ -571,3 +571,5 @@ impl PartialEq for Model {
             && self.hetero_chains == other.hetero_chains
     }
 }
+
+impl Eq for Model {}
```
Comment (nonnominandus) [inline src/structs/model.rs:575]: As Modelcontains multiple f64 values it cannot implement Eq as f64 does not implement Eq.

File: src/structs/residue.rs [hunk 4]
```diff
@@ -339,3 +339,5 @@ impl PartialEq for Residue {
             && self.atoms == other.atoms
     }
 }
+
+impl Eq for Residue {}
```
Comment (nonnominandus) [inline src/structs/residue.rs:343]: As Residue contains multiple f64 values it cannot implement Eq as f64 does not implement Eq.

**Review by nonnominandus — CHANGES_REQUESTED** [review]
I agree with the automatic derivation of clone and partial Eq for TransformationMatrix. I guess I was accustomed to manually deriving those. But I have to disagree with the Eq implementations, as f64 does not fulfil the invariants needed to support `Eq` (instead of `PartialEq`) (see this question)[https://users.rust-lang.org/t/what-is-the-difference-between-eq-and-partialeq/15751]. So Atom can never support `Eq` so all structs containing an Atom can never support `Eq`. If you remove the Eq implementations I will merge this pull request.

**Comment by TianyiShi2001** [comment]
You're correct, I didn't notice them. I've removed them.

**Review by nonnominandus — APPROVED** [review]
Good work!

**Pull request merged**